interfaces/builtin/opengl.go

```diff
@@ -74,6 +74,8 @@ unix (send, receive) type=dgram peer=(addr="@nvidia[0-9a-f]*"),
 
 # VideoCore/EGL (shared device with VideoCore camera)
 /dev/vchiq rw,
+# VideoCore Video decoding (required for accelerated MMAL video playback)
+/dev/vcsm-cma rw,
```

Comment:
don't we need udev rule for this as well? on my pi3 with groovy server installed, I see this from udevadm:

```
DEVPATH=/devices/virtual/misc/vcsm-cma
DEVNAME=/dev/vcsm-cma
DEVMODE=0666
MAJOR=10
MINOR=60
SUBSYSTEM=misc
```

which afaict won't be included in the devices cgroup when one goes to actually use this interface since none of the existing rules for this interface will match that.

Reply:
well, looking at how /dev/vchiq was originally added here, there is/was no udev rule specific to the opengl interface (there was one in the same commit for the camera interface though):

https://github.com/snapcore/snapd/commit/62d5e064fbd3b62445f03b953b37a0518886d5ad

Reply:
I'm don't think that's accurate, since the original PR adding the udev backend to the opengl interface was https://github.com/snapcore/snapd/pull/3617, and it's clearly there in the udev specs: https://github.com/snapcore/snapd/pull/3617/files#diff-92735206294aa5af0cec2a1a6d719073df872bc529c4f623b9fefb416094cff5R69

and it is still there today:

https://github.com/snapcore/snapd/blob/5d9709ae97f4414efc29b53fd19d4532be30c588/interfaces/builtin/opengl.go#L132-L134


I think the right rule for this is probably just

```go
var openglConnectedPlugUDev = []string{
...
	`KERNEL=="vcsm-cma"`,
...
```

Reply:
bah, git blame cheated me ... i'll add the rule

Reply:
the unit test needs to be updated too, I tried to push the changes but it seems I don't have perms to modify your branch perhaps because you used master, but anyways this patch should be enough:

```diff
Author: Ian Johnson <ian.johnson@canonical.com>
Date:   Mon Jan 4 21:28:43 2021 -0600

    interfaces/builtin/opengl_test.go: fix unit test
    
    Signed-off-by: Ian Johnson <ian.johnson@canonical.com>

diff --git a/interfaces/builtin/opengl_test.go b/interfaces/builtin/opengl_test.go
index e514ded9d2..9f679a0efd 100644
--- a/interfaces/builtin/opengl_test.go
+++ b/interfaces/builtin/opengl_test.go
@@ -84,7 +84,7 @@ func (s *OpenglInterfaceSuite) TestAppArmorSpec(c *C) {
 func (s *OpenglInterfaceSuite) TestUDevSpec(c *C) {
        spec := &udev.Specification{}
        c.Assert(spec.AddConnectedPlug(s.iface, s.plug, s.slot), IsNil)
-       c.Assert(spec.Snippets(), HasLen, 8)
+       c.Assert(spec.Snippets(), HasLen, 9)
        c.Assert(spec.Snippets(), testutil.Contains, `# opengl
 SUBSYSTEM=="drm", KERNEL=="card[0-9]*", TAG+="snap_consumer_app"`)
        c.Assert(spec.Snippets(), testutil.Contains, `# opengl
```

Reply:
Attempted pushing the fix, but I'm getting permission denied too. Looks like only @ogra1 can do it.

Reply:
For some reason I could push it. Once green this is good to go :)